sympy/core/add.py

```diff
@@ -1052,6 +1053,8 @@ def _eval_as_leading_term(self, x, logx=None, cdir=0):
                 n0 = min.getn()
             except NotImplementedError:
                 n0 = 1
+            if sympify(n0).has(Symbol):
```

Comment:
Will sympify be needed, if `n0 = S.One` on the line above?

Reply:
No like we need this line for the latest limit we've fixed here `limit((n+1)**k/((n+1)**(k+1) - (n)**(k+1)), n, oo)`.

In this in a intermediate step we get `min = O(n**(-k - 1))` , therefore `n0 = -k - 1` and we don't enter the `except` block during this operation .

Reply:
Maybe we could try using `sympify` in this line of `getn` function.
```
--- a/sympy/core/expr.py
+++ b/sympy/core/expr.py
@@ -1256,7 +1256,7 @@ def getn(self):
             if o.is_Symbol:
                 return S.One
             if o.is_Pow:
-                return o.args[1]
+                return sympify(o.args[1])
```
Other answers returned by the getn function are mostly sympified, so maybe we could try using this here .

Reply:
Shouldn't `o.args[1]` always be a Basic object when `o.is_Pow`?

Reply:
yes correct , that might not be the best thing to do then . I'll have a look if we can do something better than what we have at present which is `if sympify(n0).has(Symbol):`